osu.Game/Rulesets/Difficulty/DifficultyCalculator.cs

```diff
@@ -119,15 +120,23 @@ public List<TimedDifficultyAttributes> CalculateTimed([NotNull] IEnumerable<Mod>
         /// <summary>
         /// Calculates the difficulty of the beatmap using all mod combinations applicable to the beatmap.
         /// </summary>
+        /// <remarks>
+        /// This should only be used to compute difficulties for legacy mod combinations.
```

Comment:
Just to confirm my understanding - due to the overall direction of lazer (no more ranked mods => number of combination increases exponentially) this method is mostly soft-deprecated anyway and shouldn't be used for anything serious going forward, right? In that case I think this solution is fine and probably even more correct than the initial one you gave, although it's not entirely clear to me what that one would entail from the description of that.

I'd maybe even propose renaming to `CalculateAllLegacyCombinations()` or something. It's a bit annoying  to make happen as it'll be a breaking change, but I think that name yells at whoever may want to latch onto this a bit more clearly that this is probably not the method they're looking for in the long run?

Reply:
I can get behind the rename if the intentions match as stated.

Reply:
> Just to confirm my understanding

Your understanding is correct.

In the future we may use a function similar to this to compute a cache of the most common mod combinations/settings in order to not overload `osu-beatmap-difficulty-lookup-cache`, but I don't think it's going to be this method as it currently exists.